DataFormats/SoATemplate/interface/SoACommon.h

```diff
@@ -0,0 +1,610 @@
+/*
+ * Definitions of SoA common parameters for SoA class generators
+ */
+
+#ifndef DataStructures_SoACommon_h
+#define DataStructures_SoACommon_h
+
+#include <cstdint>
+#include <cassert>
+#include <ostream>
+#include <tuple>
+#include <type_traits>
+
+#include <boost/preprocessor.hpp>
+
+#include "FWCore/Utilities/interface/typedefs.h"
+
+// CUDA attributes
+#ifdef __CUDACC__
+#define SOA_HOST_ONLY __host__
+#define SOA_DEVICE_ONLY __device__
+#define SOA_HOST_DEVICE __host__ __device__
+#define SOA_HOST_DEVICE_INLINE __host__ __device__ __forceinline__
+#else
+#define SOA_HOST_ONLY
+#define SOA_DEVICE_ONLY
+#define SOA_HOST_DEVICE
+#define SOA_HOST_DEVICE_INLINE inline
+#endif
+
+// Exception throwing (or willful crash in kernels)
+#if defined(__CUDACC__) && defined(__CUDA_ARCH__)
+#define SOA_THROW_OUT_OF_RANGE(A) \
+  {                               \
+    printf(A "\n");               \
+    *((char*)nullptr) = 0;        \
+  }
+#else
+#define SOA_THROW_OUT_OF_RANGE(A) \
+  { throw std::out_of_range(A); }
+#endif
+
+/* declare "scalars" (one value shared across the whole SoA) and "columns" (one value per element) */
+#define _VALUE_TYPE_SCALAR 0
+#define _VALUE_TYPE_COLUMN 1
+#define _VALUE_TYPE_EIGEN_COLUMN 2
+
+/* The size type need to be "hardcoded" in the template parameters for classes serialized by ROOT */
+#define CMS_SOA_BYTE_SIZE_TYPE size_t
```

Comment:
Answering to myself: a type alias in the global namespace would work, but one inside a namespace would break the creation of the ROOT dictionaries :-/